internal/indexmanagement/reconcile.go

```diff
@@ -65,13 +65,7 @@ type rolloverConditions struct {
 func RemoveCronJobsForMappings(apiclient client.Client, cluster *apis.Elasticsearch, mappings []apis.IndexManagementPolicyMappingSpec, policies apis.PolicyMap) error {
 	expected := sets.NewString()
 	for _, mapping := range mappings {
-		policy := policies[mapping.PolicyRef]
-		if policy.Phases.Hot != nil {
-			expected.Insert(fmt.Sprintf("%s-rollover-%s", cluster.Name, mapping.Name))
-		}
-		if policy.Phases.Delete != nil {
-			expected.Insert(fmt.Sprintf("%s-delete-%s", cluster.Name, mapping.Name))
-		}
```

Comment:
doesn't this mean we won't remove the old cronjobs that we no longer support?

Reply:
This code produces a set of cronjobs we expect.  It will list those deployed and then remove those which are not in the expected set.  I had the same concerns but observed the operator removing the old jobs.  It would be worth QE (@anpingli)  verifying this to be the case